src/text_recognition.py

```diff
@@ -44,36 +44,71 @@ def prepare_tr_training_data(text_windows_path, n_text_windows_path):
 
     X = []
 
-    X = add_feature_data(text_windows, X)
-    X = add_feature_data(ntext_windows, X)
-
-    n_text = len(text_windows)
-    n_ntext = len(ntext_windows)
+    n_text = 0
+    for window in text_windows:
+        w = np.load(window)
+        # check for right dimensionality
+        if w.shape == (3, 3, config.NUM_D):
+            X.append(np.array(w).flatten())
+            n_text += 1
+        else:
+            pass
+
+    n_ntext = 0
+    for window in ntext_windows:
+        w = np.load(window)
+        # check for right dimensionality
+        if w.shape == (3, 3, config.NUM_D):
+            X.append(np.array(w).flatten())
+            n_ntext += 1
+        else:
+            pass
 
     y = [0] * (n_text + n_ntext)
 
-    y[0:n_text] = [1]*n_text
+    y[0:n_text] = [1] * n_text
 
     return shuffle(np.array(X), np.array(y))
 
-
 # train model
 def train_tr_model(X, y, verbose = 0):
 
-    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20, random_state=7)
+    # train test split
+    X_train, X_test, y_train, y_test = train_test_split(X[0:1000], y[0:1000], test_size=0.20, random_state=7)
```

Comment:
well, better for testing purposes, trained the model once with 40k samples and a dictionary with 500 entries - took about 4 hours :S